### PR TITLE
bug: Added a day to the settlement download input parameters

### PR DESCRIPTION
**Pull request by energinet-ajw**
Because the end period in the Wholesale API is exclusive, so we need to add one day to the period.
Example - Before the fix.

In UI we select the period 2023-11-14 to 2023-11-15
This is converted to 2023-11-13 to 2023-11-14 before requesting the Wholesale API because of a UTC conversion and a substring operation. And this is the wrong period, that is, one day back in time.

https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/team-mandalorian/80

Reach out if you have a better solution.

### CODE REVIEW

File: libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts [hunk 1]
```diff
@@ -218,12 +219,15 @@ export class DhWholesaleSettlementsReportsTabsBalanceComponent
       message: this.transloco.translate('wholesale.settlementReports.downloadStart'),
     });
 
+    const startPeriod = this.tryAddOneDay(this.searchForm.controls.period.value?.start);
```
Comment (frontend-specialisten) [inline libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts:222]: Question: I understand what you're doing, however my question is if we're interested in calling the API if the periods  are not set? (the empty string fallback).

File: libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts [hunk 2]
```diff
@@ -237,4 +241,12 @@ export class DhWholesaleSettlementsReportsTabsBalanceComponent
           }),
       });
   }
+
+  private tryAddOneDay(value: unknown): unknown {
```
Comment (frontend-specialisten) [inline libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts:245]: Question: From what I can see, the value should be of type either `string | undefined`, so I can't see how your if statement is ever being executed?

File: libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts (hunk 1 above)
Comment (energinet-ajw) [reply libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts:222]: Good question. The previous code had this logic:

this.searchForm.controls.period.value?.start ?? ''

File: libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts (hunk 2 above)
Comment (energinet-ajw) [reply libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts:245]: I can't use string or undefined.

**Review by frontend-specialisten — APPROVED** [review]

**Pull request merged**